src/ios/addToHeaderSearchPaths.js

```diff
@@ -1,29 +1,5 @@
-/**
- * Given Xcode project and path, iterate over all build configurations
- * and append new path to HEADER_SEARCH_PATHS section
- *
- * We cannot use builtin addToHeaderSearchPaths method since react-native init does not
- * use $(TARGET_NAME) for PRODUCT_NAME, but sets it manually so that method will skip
- * that target.
- *
- * To workaround that issue and make it more bullet-proof for different names,
- * we iterate over all configurations and look if React is already there. If it is,
- * we assume user will need newly added dependency headers in it as well.
- */
-module.exports = function addToHeaderSearchPaths(project, path) {
-  const config = project.pbxXCBuildConfigurationSection();
-
-  Object
-    .keys(config)
-    .filter(ref => ref.indexOf('_comment') === -1)
-    .forEach(ref => {
-      const buildSettings = config[ref].buildSettings;
-      const shouldAddSearchPath = (buildSettings.HEADER_SEARCH_PATHS || [])
-        .filter(path => path.indexOf('react-native/React/**'))
-        .length > 0;
+const mapHeaderSearchPaths = require('./mapHeaderSearchPaths');
 
-      if (shouldAddSearchPath) {
-        buildSettings.HEADER_SEARCH_PATHS.push(path);
-      }
-    });
+module.exports = function addToHeaderSearchPaths(project, path) {
+  mapHeaderSearchPaths(project, searchPaths => searchPaths.concat(path));
```

Comment:
Usually you pass an array to map, here we pass project and mapping happens inside it (check the `func` application. It's mutating because of underyling library design, so the only way to get over that was to keep that in a separate stuff.